src/AWS.Deploy.Common/DeploymentManifest/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+
+namespace AWS.Deploy.Common.DeploymentManifest
+{
+    public interface IDeploymentManifestEngine
+    {
+        string? TargetApplicationFullPath { get; set; }
+
+        Task UpdateDeploymentManifestFile(string saveCdkDirectoryFullPath);
+    }
+
+    /// <summary>
+    /// This class contains the helper methods to update the deployment manifest file
+    /// that keeps track of the save CDK deployment projects.
+    /// </summary>
+    public class DeploymentManifestEngine : IDeploymentManifestEngine
+    {
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+
+        public string? TargetApplicationFullPath { get; set; }
```

Comment:
I don't see why this is a public variable. This would be better passed as a parameter to UpdateDeploymentManifestFile. That way, you don't need to define it as a nullable string and set it after initializing the deployment manifest engine.

Reply:
Fixed. Passing it as parameter to methods which need this.